### PR TITLE
Add a nil UpdateStrategy round-trip case

### PR DESCRIPTION
**Pull request by shrutiyam-glitch**
#### What this PR does / why we need it:

This PR adds a conversion test case to verify that `SandboxWarmPool` conversions round-trip correctly when the `UpdateStrategy` is `nil`. 

By adding this test case to `extensions/api/v1alpha1/sandboxwarmpool_conversion_test.go`, we ensure that API version conversions handle `nil` or empty update strategies gracefully, preventing potential regressions or nil-pointer dereferences during conversion round-trips.

#### Which issue(s) this PR is related to:

Based on this comment - https://github.com/kubernetes-sigs/agent-sandbox/pull/993#pullrequestreview-4546773559 

## Summary by CodeRabbit

* **Tests**
  * Improved coverage for sandbox warm pool conversion edge cases, adding table-driven subtests to verify correct round-trip behavior for optional update strategy configuration (including `nil` handling) and ensuring warm-pool state annotations are stripped as expected.

### CODE REVIEW

File: extensions/api/v1alpha1/sandboxwarmpool_conversion_test.go [hunk 1]
```diff
@@ -120,3 +120,39 @@ func TestSandboxWarmPoolConversion(t *testing.T) {
 		t.Errorf("roundtrip Selector mismatch: expected %q, got %q", src.Status.Selector, roundTrip.Status.Selector)
 	}
 }
+
+// TestSandboxWarmPoolConversion_NilUpdateStrategy exercises the nil branch of
```
Comment (janetkuo) [inline extensions/api/v1alpha1/sandboxwarmpool_conversion_test.go]: This duplicates basic setup but misses the rigorous metadata assertions in the existing TestSandboxWarmPoolConversion. The new test case should be folded into TestSandboxWarmPoolConversion by refactoring it into a table-driven test.

**Review by janetkuo — APPROVED** [review]
The table-driven refactoring is clean and correct.

/lgtm
/approve

**Review by janetkuo — APPROVED** [review]
/lgtm
